src/main/java/com/maxdemarzi/shortest/Service.java

```diff
@@ -438,6 +415,96 @@ private void streamShortestPathsUsingHandwrittenBFS(String centerEmail, List<Str
+                .build();
+            Builder<Integer, Integer> builder = ImmutableMap.<Integer, Integer>builder();
+            for (Map.Entry<String, Integer> e : costs.entrySet()) {
+                builder.put(readOps.relationshipTypeGetForName(e.getKey()), e.getValue().intValue());
+            }
+            relationships = HashIntIntMaps.getDefaultFactory()
+                .withDefaultValue(100) // makes it too expensive to explore relationships not listed here
+                .newImmutableMap(builder.build());
+        }
+        return relationships;
+    }
+
+    private void streamShortestPathsUsingDijkstra(String centerEmail, List<String> bibEntries, List<String> edgeEmails, int maxLength, JsonGenerator jg) {
+        try (Transaction tx = db.beginTx()) {
+            final Long centerNodeId;
+            try {
+                centerNodeId = nodeCache.getEmailNode(centerEmail);
+            } catch (ExecutionException e) {
+                return;
+            }
+            Map<Long, Integer> startNodes = HashLongIntMaps.newMutableMap();
+            for (Long nodeId : nodeCache.getBibliographEntryNodes(bibEntries))  {
+                startNodes.put(nodeId, 1);
+            }
+            startNodes.put(centerNodeId, 0);
+
+            final HashLongObjMap<String> edgeEmailsByNodeId = HashLongObjMaps.newMutableMap();
+            for (String edgeEmail : edgeEmails) {
+                try {
+                    edgeEmailsByNodeId.put(nodeCache.getEmailNode(edgeEmail), edgeEmail);
+                } catch (Exception e) {
+                    continue;
+                }
+            }
+
+            ThreadToStatementContextBridge ctx = dbAPI.getDependencyResolver().resolveDependency(ThreadToStatementContextBridge.class);
+            ReadOperations ops = ctx.get().readOperations();
+
+            Dijkstra dijkstra = new Dijkstra(ops, relationshipCosts(ops), startNodes, maxLength, new Traversal.NodeCallback() {
+                public void explored(Traversal traversal, NodeItem node, long nodeId, int cost, int paths) {
+                    String email = edgeEmailsByNodeId.remove(nodeId);
+                    if (email != null) { //found a match!
+                        try {
+                            writeResultObject(jg, email, cost, paths);
+                        } catch(IOException ex) {
+                            traversal.finish();
+                            return;
+                        }
+                        if (edgeEmailsByNodeId.isEmpty()) {
+                            traversal.finish();
+                        }
+                    }
+                }
+            });
+            dijkstra.run();
+            for (Long nodeId : edgeEmailsByNodeId.keySet()) {
+                new OneDegreeTraversal(ops, relationshipCosts(ops), nodeId.longValue(), 0, new Traversal.NodeCallback() {
+                    public void explored(Traversal traversal, NodeItem node, long connectingNode, int cost, int paths) {
```

Comment:
should be changed if possible to get an accurate path count, so we explore all relationships for a given edge node, counting paths, before returning it